CMakeLists.txt

```diff
@@ -4,6 +4,12 @@ project(${project_name})
 
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}  -pthread -std=c++17")
 
+if(ENABLE_ASIO_STANDALONE)
+	add_definitions(-DASIO_STANDALONE)
+else()
+	find_package(Boost 1.60 REQUIRED COMPONENTS system)
```

Comment:
How about downloading boost dynamically? So that the users shouldn't concern what he/she should do to prepare the dependencies.

You can refer this https://github.com/jovany-wang/dousi/blob/master/cpp/cmake/Modules/GtestExternalProject.cmake